registry/api/v2/descriptors.go

```diff
@@ -361,6 +361,48 @@ var routeDescriptors = []RouteDescriptor{
 		},
 	},
 	{
+		Name:        RouteNameCatalog,
```

Comment:
Rather than copying this from #583, just rebase this branch on the commits from that PR branch. When everything is good to go, we can merge it as a unit.